packages/monaco/src/browser/monaco-snippet-suggest-provider.ts

```diff
@@ -196,7 +197,10 @@ export class MonacoSnippetSuggestProvider implements monaco.languages.Completion
             if (JsonSerializedSnippet.is(scopeOrTemplate)) {
                 accept(name, scopeOrTemplate);
             } else {
-                this.parseSnippets(scopeOrTemplate, accept);
+                // eslint-disable-next-line @typescript-eslint/no-shadow
+                for (const [name, template] of Object.entries(scopeOrTemplate)) {
```

Comment:
Even using the full set of new vscode built-in extensions I could not get a single one to go into this branch, thanks to the other changes in this PR. I think there's similar handling in vscode so I'm ok to keep it untested. 

The important thing is that you replaced the recursion, so at worst the damages will be limited if we have issues in the future e.g. because a new type of snippet has been added that needs special handling.

Reply:
Yeah, this part is pretty much a copy of what I found in VS Code. Thanks for telling me that it was pretty much impossible to get anything through that branch.